artemis-server/src/main/java/org/apache/activemq/artemis/core/deployers/impl/FileConfigurationParser.java

```diff
@@ -768,6 +772,64 @@ public void parseMainConfig(final Element e, final Configuration config) throws
       }
    }
 
+
+   private void parseJournalRetention(final Element e, final Configuration config) {
+      NodeList retention = e.getElementsByTagName("journal-retention-directory");
+
+      if (retention.getLength() != 0) {
+         Element node = (Element) retention.item(0);
+
+         String directory = node.getTextContent().trim();
+
+         String storageLimitStr = getAttributeValue(node, "storage-limit");
+         long storageLimit;
+
+         if (storageLimitStr == null) {
+            storageLimit = -1;
+         } else {
+            storageLimit = ByteUtil.convertTextBytes(storageLimitStr.trim());
+         }
+         int period = getAttributeInteger(node, "period", -1, Validators.GT_ZERO);
+         String unitStr = getAttributeValue(node, "unit");
+
+         if (unitStr == null) {
+            unitStr = "DAYS";
+         }
+
+         TimeUnit unit;
+
+         switch (unitStr) {
```

Comment:
Why not use TimeUnit.valueOf its already an enum?

Reply:
e.g. this switch could be simplified down to:

unit = TimeUnit.valueOf(unitStr.toUpperCase());

Reply:
Nice.. idea.. done!